public class org.springframework.core.NestedIOException extends java.io.IOException
{

    static void <clinit>()
    {
        java.lang.Class $r0;

        $r0 = class "Lorg/springframework/core/NestedExceptionUtils;";

        virtualinvoke $r0.<java.lang.Class: java.lang.String getName()>();

        return;
    }

    public void <init>(java.lang.String)
    {
        org.springframework.core.NestedIOException r0;
        java.lang.String r1;

        r0 := @this: org.springframework.core.NestedIOException;

        r1 := @parameter0: java.lang.String;

        specialinvoke r0.<java.io.IOException: void <init>(java.lang.String)>(r1);

        return;
    }

    public void <init>(java.lang.String, java.lang.Throwable)
    {
        java.lang.Throwable r2;
        org.springframework.core.NestedIOException r0;
        java.lang.String r1;

        r0 := @this: org.springframework.core.NestedIOException;

        r1 := @parameter0: java.lang.String;

        r2 := @parameter1: java.lang.Throwable;

        specialinvoke r0.<java.io.IOException: void <init>(java.lang.String)>(r1);

        virtualinvoke r0.<org.springframework.core.NestedIOException: java.lang.Throwable initCause(java.lang.Throwable)>(r2);

        return;
    }

    public java.lang.String getMessage()
    {
        java.lang.Throwable $r2;
        org.springframework.core.NestedIOException r0;
        java.lang.String $r1, $r3;

        r0 := @this: org.springframework.core.NestedIOException;

        $r1 = specialinvoke r0.<java.io.IOException: java.lang.String getMessage()>();

        $r2 = virtualinvoke r0.<org.springframework.core.NestedIOException: java.lang.Throwable getCause()>();

        $r3 = staticinvoke <org.springframework.core.NestedExceptionUtils: java.lang.String buildMessage(java.lang.String,java.lang.Throwable)>($r1, $r2);

        return $r3;
    }
}
